src/semaphore.ts

```diff
@@ -19,10 +19,6 @@ export class Semaphore {
     return this._queue.length;
   }
 
-  get acquired() {
-    return this._maxAllowed - this._permits;
-  }
-
   public async acquire(): Promise<void> {
     if (this._permits > 0) {
       this._permits -= 1;
```

Comment:
### Code Review

#### Bugs / Risks:
- The removal of the `acquired` getter method decreases visibility into the state of the Semaphore, which might be useful for debugging or auditing Semaphore usage. Clients relying on this method will break after updating.

#### Improvements:
- The logic within `acquire()` assumes that there are permits available (`this._permits > 0`), but it does not handle the case where `_permits` equals `0`, potentially leading to underflow if `acquire()` is called mistakenly when `_permits` is already `0`.
- The asynchronous nature of `acquire()` implies it should handle waiting for a permit to become available. However, without seeing the rest of the code, it's impossible to know if it properly queues requests and resolves them later when permits are released.

#### Security Vulnerabilities: 
- From the provided snippet, there's no direct indication of security issues, but improper management of semaphore state could indirectly lead to conditions where security controls expecting mutual exclusion don't behave as expected.

**Rate**: Given the lack of context, the potential bug risks tied to state management and the incomplete implementation of `acquire()`, I would rate this code snippet a 4 out of 10.

Quality Rate: 4/10